tests/test_permissions.py

```diff
@@ -0,0 +1,151 @@
+from pyartifactory.models.Permission import Permission, SimplePermission
+
+URL = "http://localhost:8080/artifactory"
+AUTH = ("user", "password_or_apiKey")
+
+SIMPLE_PERMISSION = SimplePermission(name="test_permission", uri="someuri")
+PERMISSION = Permission(
+    **{
+        "name": "test_permission",
+        "repositories": ["test_repository"],
+        "principals": {
+            "users": {"test_user": ["r", "w", "n", "d"]},
+            "groups": {"developers": ["r"]},
+        },
+    }
+)
+
+
+class TestGroup:
+    @staticmethod
+    @responses.activate
+    def test_create_permission_fail_if_group_already_exists(mocker):
+        responses.add(
+            responses.GET,
+            f"{URL}/api/security/permissions/{PERMISSION.name}",
+            json=PERMISSION.dict(),
+            status=200,
+        )
+
+        artifactory_permission = ArtfictoryPermission(AuthModel(url=URL, auth=AUTH))
+        mocker.spy(artifactory_permission, "get")
+        with pytest.raises(PermissionAlreadyExistsException):
+            artifactory_permission.create(PERMISSION)
+
+        artifactory_permission.get.assert_called_once_with(PERMISSION.name)
+
+    @staticmethod
+    @responses.activate
+    def test_create_permission_success(mocker):
+        responses.add(
+            responses.GET,
+            f"{URL}/api/security/permissions/{PERMISSION.name}",
+            json=PERMISSION.dict(),
+            status=404,
+        )
+        responses.add(
+            responses.PUT,
+            f"{URL}/api/security/permissions/{PERMISSION.name}",
+            json=PERMISSION.dict(),
+            status=201,
+        )
+
+        artifactory_permission = ArtfictoryPermission(AuthModel(url=URL, auth=AUTH))
+        mocker.spy(artifactory_permission, "get")
+        with pytest.raises(PermissionNotFoundException):
+            artifactory_permission.create(PERMISSION)
+
+            artifactory_permission.get.assert_called_with(PERMISSION.name)
+        assert artifactory_permission.get.call_count == 2
+
+    @staticmethod
+    @responses.activate
+    def test_get_permission_error_not_found():
+        responses.add(
+            responses.GET,
+            f"{URL}/api/security/permissions/{PERMISSION.name}",
+            status=404,
+        )
+
+        artifactory_permission = ArtfictoryPermission(AuthModel(url=URL, auth=AUTH))
+        with pytest.raises(PermissionNotFoundException):
+            artifactory_permission.get(PERMISSION.name)
+
+    @staticmethod
+    @responses.activate
+    def test_get_permission_success(mocker):
+        responses.add(
+            responses.GET,
+            f"{URL}/api/security/permissions/{PERMISSION.name}",
+            json=PERMISSION.dict(),
+            status=200,
+        )
+
+        artifactory_permission = ArtfictoryPermission(AuthModel(url=URL, auth=AUTH))
+        mocker.spy(artifactory_permission, "get")
+        artifactory_permission.get(PERMISSION.name)

```

Comment:
You might want to assert that the object generated from this API call is the same as the one sent by the API (this applies to many other tests in this file).